### DISCUSSION TITLE
change port database

### DISCUSSION THREAD

**Discussion by bff0172** [pkuehnel/TeslaSolarCharger#1135] [Q&A]
I want to run tsc on Synology NAS with docker installed. There is already a postgres database for internal purposes which cannot be used for own applications. How can I change the yml to use another tcp port for pgsql?

**Comment by pkuehnel** [comment]
I stead of
5432:5432 use e.g. 5433:5432

As TeslaMate and TSC itself use an internal port the change does not matter.

**Comment by bff0172** [comment]
thanks, but when I insert this to yml:
db:
image: postgres:11
environment:
- POSTGRES_USER=teslamate
- POSTGRES_PASSWORD=xxx
ports:
- 5433:5432
volumes:
- teslamate-db:/var/lib/postgresql/data

I got after docker-compose up:
ERROR: In file './docker-compose.yml', service must be a mapping, not a NoneType.

**Reply by pkuehnel** [reply]
The spaces must be correct. Can't tell you the issue, as the format in your message is not the same as it is in the yaml. Try using codeblocks here to preserve the text format when pasting the yml code on Github. Why are you using Postgres version 11? is it a new installation you create here, or did you use TeslaMate before?

**Reply by bff0172** [reply]
the installation is from September, shall I use an older one?

**Reply by bff0172** [reply]
sorry, no I use version 15 - the tag is ll - so last

**Reply by pkuehnel** [reply]
As upgrading is not easy and it should work with version 11 as well: not necessarily but you should have used version 15 back in September ;-)

**Reply by bff0172** [reply]
you're rigth, this was not the right yml - I have 15

**Reply by bff0172** [reply]
Thanks, now it works with 5433 - and my local pgsql with 5432 - so if somebody need help for installing tsc on Synology NAS can contact me.
br
